src/models/task.rs

```diff
@@ -407,7 +409,8 @@ mod tests {
 
         assert_eq!(task.spec.resources.cpus.millicores(), Ok(1000));
         assert_eq!(task.spec.resources.gpus.millicores(), Ok(1000));
-        assert_eq!(task.spec.resources.memory.bytes(), Ok(1024));
+        // task resources are specified with ByteUnit<DefaultFactorOneMegabyte>::Number(1024)
+        assert_eq!(task.spec.resources.memory.bytes(), Ok(1024 * 1024 * 1024));
```

Comment:
Assert in `src/models/serialization_helpers.rs:60` suggests that in our API `MB` means SI megabytes, i.e. power of 10. Here we have megabytes with powers of 2 if I understand this correctly.

I think we should align these. Otherwise the named label for unit can be ambiguous and effectively lose meaning.

Reply:
That's the issue I mentioned in questions in #51. The situation now is the following:

- When using deserialization (e.g. reading a `task.yaml` file or `FromStr`):
  - string values will be parsed as `bytesize` crate does it - with respect to SI: `"1MB"` is 10^6 bytes, `"1MiB"` is 2^20 bytes
  - integer values will be parsed as mebibytes: `1` is 2^20 bytes.
- When using API:
  - `ByteUnit<DefaultFactorOneMegabyte>` is actually one mebibyte, so `ByteUnit<DefaultFactorOneMegabyte>::Number(1)` is 2^20 bytes.

This is a bit confusing, I agree. Should we do something about that right now?

Reply:
> Should we do something about that right now?
Yes, I think `DefaultFactorOneMegabyte` should be actual SI megabytes (10^6) if we are supporting both formats w/ string parsing.

So therefore `ByteUnit<DefaultFactorOneMegabyte>::Number(1)` should yield 10^6 bytes, instead of 2^20.

Wouldn't that align everything correctly?

Reply:
This PR addresses the issues:
https://github.com/gevulotnetwork/gevulot-rs/pull/57